src/QueryStrategies/BatchQueryStrategies/batch_qs_base.jl

```diff
@@ -0,0 +1,130 @@
+abstract type BatchPQs <: PoolQs end
+abstract type ExtendingBatchQs <: BatchPQs end
+abstract type MultiObjectiveBatchQs <: BatchPQs end
+
+"""
+    select_batch(batch_query_strategy, query_data, label_map, candidate_indices)
+Applies batch_query_strategy to select the most useful unlabeled observations
+from query_data.
+
+label_map is used to find out which observations in query_data are still unlabeled,
+candidate_indices is used to find out which observations in query_data are valid
+candidates for the query.
+
+Returns list of indices of most useful observations.
+"""
+function select_batch end
+
+function get_query_objects(qs::BatchPQs,
+                           query_data::Array{T, 2} where T <: Real,
+                           pools::Vector{Symbol},
+                           global_indices::Vector{Int},
+                           history::Vector{Vector{Int}})::Vector{Int}
+    pool_map = MLLabelUtils.labelmap(pools)
+    haskey(pool_map, :U) || throw(ArgumentError("No more points that are unlabeled."))
+    all_history_values = (length(history) > 0) && (collect(Iterators.flatten(history)))
```

Comment:
This is type instable. If `(length(history) > 0)` evaluates to false, `all_history_values` is a boolean, otherwise an Int[]. This should just be 

```julia 
all_history_values = collect(Iterators.flatten(history))
```
which returns a flat empty list if history is empty.

Reply:
I agree. As a side note, it's surprising that `3 in false` works in Julia and returns `false`.